sqlglot/generator.py

```diff
@@ -3757,3 +3757,53 @@ def _replace_line_breaks(self, string: str) -> str:
         if self.pretty:
             return string.replace("\n", self.SENTINEL_LINE_BREAK)
         return string
+
+    def copyparameter_sql(self, expression: exp.CopyParameter) -> str:
+        option = self.sql(expression, "this")
+        value = self.sql(expression, "value")
+        value = f" {value}" if value else ""
+
+        return f"{option}{value}"
+
+    def copy_sql(self, expression: exp.Copy) -> str:
+        this = self.sql(expression, "this")
+        this = f" INTO {this}" if expression.args.get("into") else f" {this}"
+
+        credentials = self.sql(expression, "credentials")
+        credentials = f" {credentials}" if credentials else ""
+
+        kind = " FROM " if expression.args.get("kind") else " TO "
+        files = self.expressions(expression, "files", flat=True)
+
+        sep = ", " if self.dialect.COPY_PARAMS_SEP else " "
+        params = self.expressions(expression, "params", flat=True, sep=sep)
+        if params:
+            params = self.wrap(params) if expression.args.get("wrapped") else params
+            params = f" WITH {params}" if expression.args.get("with_token") else f" {params}"
+
+        return f"COPY{this}{kind}{files}{credentials}{params}"
+
+    def credentials_sql(self, expression: exp.Credentials) -> str:
+        cred_expr = expression.args.get("credentials")
+        if isinstance(cred_expr, exp.Literal):
+            # Redshift case: CREDENTIALS <string>
+            credentials = self.sql(expression, "credentials")
+            credentials = f"CREDENTIALS {credentials}"
+        else:
+            # Snowflake case: CREDENTIALS = (...)
+            credentials = self.expressions(expression, "credentials", flat=True, sep=" ")
+            credentials = f"CREDENTIALS = ({credentials})" if credentials else ""
```

Comment:
Should we factor this out into a helper method? Eg. `generated_credentials`, which generates the "snowflake" case by default and then can be overridden in Redshift. Is Redshift the only one with this syntax?

Reply:
Both dialects have unique syntax, so I thought that prefering one over the other might not be semantically correct and that there's not much to gain from that extra fragmentation. However, either approach works for me